client/lib/media/test/actions.js

```diff
@@ -184,91 +192,113 @@ describe( 'MediaActions', function() {
+				} );
+
+				done();
 			} );
 		} );
 
-		it( 'should accept a plain object file descriptor', function() {
+		it( 'should accept a plain object file descriptor', function( done ) {
 			var file = { file: DUMMY_UPLOAD, parent_id: 300 };
 			sandbox.stub( PostEditStore, 'get' ).returns( { ID: 200 } );
 
-			return MediaActions.add( DUMMY_SITE_ID, file ).then( () => {
+			return MediaActions.add( DUMMY_SITE_ID, file, { waitForUpload: true } ).then( () => {
 				expect( mediaAdd ).to.have.been.calledWithMatch( {}, file );
+
+				done();
 			} );
 		} );
 
-		it( 'should call to the WordPress.com REST API', function() {
-			return MediaActions.add( DUMMY_SITE_ID, DUMMY_UPLOAD ).then( () => {
+		it( 'should call the WordPress.com REST API', function( done ) {
+			return MediaActions.add( DUMMY_SITE_ID, DUMMY_UPLOAD, { waitForUpload: true } ).then( () => {
 				expect( mediaAdd ).to.have.been.calledWithMatch( {}, DUMMY_UPLOAD );
 				expect( Dispatcher.handleServerAction ).to.have.been.calledWithMatch( {
 					type: 'RECEIVE_MEDIA_ITEM',
 					siteId: DUMMY_SITE_ID,
 					id: 'media-1',
 					data: DUMMY_API_RESPONSE.media[ 0 ]
 				} );
+
+				done();
 			} );
 		} );
 
-		it( 'should immediately receive a transient object', function() {
-			MediaActions.add( DUMMY_SITE_ID, DUMMY_UPLOAD );
+		it( 'should immediately receive a transient object', function( done ) {
+			MediaActions.add( DUMMY_SITE_ID, DUMMY_UPLOAD ).then( () => {
+				expect( Dispatcher.handleViewAction ).to.have.been.calledWithMatch( {
+					type: 'CREATE_MEDIA_ITEM',
+					data: {
+						ID: 'media-1',
+						file: DUMMY_UPLOAD.name,
+						'transient': true
+					}
+				} );
 
-			expect( Dispatcher.handleViewAction ).to.have.been.calledWithMatch( {
-				type: 'CREATE_MEDIA_ITEM',
-				data: {
-					ID: 'media-1',
-					file: DUMMY_UPLOAD.name,
-					'transient': true
-				}
+				done();
 			} );
 		} );
 
-		it( 'should attach file upload to a post if one is being edited', function() {
+		it( 'should attach file upload to a post if one is being edited', function( done ) {
 			sandbox.stub( PostEditStore, 'get' ).returns( { ID: 200 } );
 
-			return MediaActions.add( DUMMY_SITE_ID, DUMMY_UPLOAD ).then( () => {
+			return MediaActions.add( DUMMY_SITE_ID, DUMMY_UPLOAD, { waitForUpload: true } ).then( () => {
 				expect( mediaAdd ).to.have.been.calledWithMatch( {}, {
 					file: DUMMY_UPLOAD,
 					parent_id: 200
 				} );
+
+				done();
 			} );
 		} );
 
-		it( 'should attach URL upload to a post if one is being edited', function() {
+		it( 'should attach URL upload to a post if one is being edited', function( done ) {
 			sandbox.stub( PostEditStore, 'get' ).returns( { ID: 200 } );
 
-			return MediaActions.add( DUMMY_SITE_ID, DUMMY_URL ).then( () => {
+			return MediaActions.add( DUMMY_SITE_ID, DUMMY_URL, { waitForUpload: true } ).then( () => {
 				expect( mediaAddUrls ).to.have.been.calledWithMatch( {}, {
 					url: DUMMY_URL,
 					parent_id: 200
 				} );
+
+				done();
```

Comment:
Did you find you needed to call `done` in these tests? Mocha is happy to work with promise return values in lieu of an explicit callback.